### ISSUE TITLE
unsafe precondition(s) violated: slice::from_raw_parts requires the pointer to be aligned and non-null, and the total size of the slice not to exceed `isize::MAX`

### ISSUE THREAD

**Issue by ghostman2013** [KyleMayes/vulkanalia#254]
Hello!

I tried to copy a code from [this tutorial](https://github.com/KyleMayes/vulkanalia/blob/master/tutorial/src/01_instance_creation.rs) and got the error:  

```
thread 'main' panicked at library/core/src/panicking.rs:156:5:
unsafe precondition(s) violated: slice::from_raw_parts requires the pointer to be aligned and non-null, and the total size of the slice not to exceed `isize::MAX`
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread caused non-unwinding panic. aborting.
```

It happens at **119 line** but I use `vulkanalia = { version = "0.22.0", features = ["libloading", "provisional", "window"] }`:  
```
// Create

let info = vk::InstanceCreateInfo::builder()
    .application_info(&application_info)
    .enabled_extension_names(&extensions)
    .flags(flags);

Ok(entry.create_instance(&info, None)?)
```

In `entry.create_instance()` the library calls `get_names(info.enabled_layer_count, info.enabled_layer_names);` that calls 
```
slice::from_raw_parts(strings, num_strings as usize)
        .iter()
        .map(|s| vk::ExtensionName::from_ptr(*s))
        .collect()
```

In this sample the number of layers is 0 and the array points to `nullptr`. It looks like we need to check the number of layers and just return an empty `layers: HashSet<StringArray<256>>` to `Instance`.

Or am I doing a something wrongly? And I must have layers > 0?

**Comment by ghostman2013** [comment]
On `0.21.0` the same problem

**Comment by KyleMayes** [comment]
I had trouble replicating this, but eventually I found that this assertion is currently only enabled in nightly compilers.

I was indeed providing a null pointer when no layers and/or extensions were supplied which violates the pre-conditions of `slice::from_raw_parts`.

I've fixed the which should be released in `v0.23.0`.

**Comment by ghostman2013** [comment]
> I had trouble replicating this, but eventually I found that this assertion is currently only enabled in nightly compilers.
> 
> I was indeed providing a null pointer when no layers and/or extensions were supplied which violates the pre-conditions of `slice::from_raw_parts`.
> 
> I've fixed the which should be released in `v0.23.0`.

You're right, I use a nightly compiler. Honestly, I even forgot already, I've switched several months ago. 

Thank you for the answer!
